web/src/views/InvestigationDetail.vue

```diff
@@ -151,17 +172,82 @@ export default defineComponent({
       geojsParams.layer.crossDomain = 'use-credentials';
       createMap(geojsParams.map);
       createLayer('osm', geojsParams.layer);
+      clampBoundsX(false);
+    });
+    // #endregion datasets
+
+    // #region pin
+    const selectedPins: Ref<Pin[]> = computed(() => store.state.selectedPins);
+    /* eslint-disable */
+    let featureLayer: any;
+    let pinFeature: any;
+    /* eslint-enable */
+    watch(selectedPins, (pinList) => {
+      if (!featureLayer) {
+        featureLayer = createLayer('feature', { features: ['point', 'line', 'polygon'] });
+      }
+      const pinFeatureData = pinList.map((pin) => {
+        let pinLocation: Point | undefined = postGisToPoint(pin.child_location);
+        if (!pinLocation) {
+          pinLocation = { x: 0, y: 0 };
+        }
```

Comment:
I think the following should work (but I haven't tested it):

```suggestion
        const pinLocation: Point = postGisToPoint(pin.child_location) || { x: 0, y: 0 };
```